perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -241,11 +255,68 @@ def SetFiles(self):
       self.RemoteCommand('echo "%s" | sudo tee %s' %
                          (value, path))
 
+  def ApplySysctlPersistent(self, key, value):
+    """Apply "key=value" pair to /etc/sysctl.conf and reboot.
+
+    The reboot ensures the values take effect and remain persistent across
+    future reboots.
+
+    Args:
+      key: a string - the key to write as part of the pair
+      value: a string - the value to write as part of the pair
+    """
+
+    self.RemoteCommand('sudo bash -c \'echo "%s=%s" >> /etc/sysctl.conf\''
+                       % (key, value))
+
+    self._needs_reboot = True
+
   def DoSysctls(self):
-    """Apply --sysctl to the VM."""
+    """Apply --sysctl to the VM.
+
+       The Sysctl pairs are written persistently so that if a reboot
+       occurs, the flags are not lost.
+    """
+
+    for pair in FLAGS.sysctl:
+      key, value = pair.split('=')
+      self.ApplySysctlPersistent(key, value)
+
+  def DoConfigureNetworkForBBR(self):
+    """Apply --network_enable_BBR to the VM."""
+
+    if not FLAGS.network_enable_BBR:
+      return
+
+    if not self.CheckKernelVersion().AtLeast(4, 9):
+      raise flags.ValidationError(
+          'BBR requires a linux image with kernel 4.9 or newer')
+
+    # if the current congestion control mechanism is already BBR
+    # then nothing needs to be done (avoid unnecessary reboot)
+    if self.TcpCongestionControl() == 'bbr':
+      return
+
+    self.ApplySysctlPersistent('net.core.default_qdisc', 'fq')
+    self.ApplySysctlPersistent('net.ipv4.tcp_congestion_control', 'bbr')
+
+  def _RebootIfNecessary(self):
+    """Will reboot the VM if self._needs_reboot has been set."""
+
+    if self._needs_reboot:
+      self.Reboot()
+      self._needs_reboot = False
 
-    if FLAGS.sysctl:
-      self.RemoteCommand('sudo sysctl -w %s' % (' '.join(FLAGS.sysctl),))
+  def TcpCongestionControl(self):
+    """Returns the congestion control used for tcp."""
```

Comment:
Docstrings descriptions should be written as a command. From the pep: 
"It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ..."."

Reply:
A good majority of Docstring description in this file seems to violate this rule ...  See GetSerivceName, GetPathToConfig, FormatDisk, RemoteHostCopy ....etc

Reply:
(making the change just the same ....)